lwt/client/dns_client_lwt.ml

```diff
@@ -266,6 +273,8 @@ module Transport : Dns_client.S
       Lwt.async (fun () -> Lwt_list.iter_p (handle_action t) actions);
       waiter >>= function
       | Error `Msg msg ->
+        Lwt_condition.broadcast connected_condition ();
```

Comment:
If an error happens while multiple 'threads' are trying to connect it would result in retrying immediately one by one. Ideally, there would be some back-off semantics. Perhaps the condition could propagate a value of type `` [`Connected | `Retry of timeout]``, and in the case of error use Lwt_condition.signal and reuse the condition.

Reply:
thanks for your review. My thinking is (taking non-preemptive scheduling of Lwt into account):
- we have a connection failure, and broadcast to all waiting DNS resolutions
- one of which is scheduled, which calls `connect_via_tcp_to_ns` again, which does the match on fd and connected_condition (hint: both are None)
- the branch where a new condition is allocated and t.connected_condition is mutated is taken, which leads to another connection attempt

all other tasks that are scheduled now will "wait" for yet another broadcast.